src/types/select.js

```diff
@@ -797,17 +797,17 @@ export default {
 	},
 
 	deMap: function (obj) {
-		if(!obj){
+		if (!obj) {
 			return null;
 		}
-		return obj.value || null;
+		return obj.value;
 	},
 
 	isEmpty: function () {
 		const self = this,
 			data = self.deMap(self.item);
 
-		return data !== 0 && !data;
+		return data === 0 || data === undefined || data === null || data === '';
```

Comment:
Sergio had mentioned that 0 should be empty.
The JIRA was because explicitly false values are not empty.  I need to be able to pass in a false and have it pass validation.  undefined and null are still not allowed.
AFAIK, the empty values are enumerated now.  All other values should be considered not empty.